File: app/src/main/java/jgappsandgames/smartreminderslite/holder/TagHolder.java
package jgappsandgames.smartreminderslite.holder;

// Views
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.TextView;

// App
import jgappsandgames.smartreminderslite.R;

/**
 * TagHolder
 * Created by joshua on 8/31/17.
 */
public class TagHolder implements OnClickListener {
    // Data
    private final TagSwitcher switcher;

    private boolean tag_selected;
    private String tag_text;

    // Views
    private final TextView text_view;

    // Initializer
    public TagHolder(String tag, boolean selected, TagSwitcher switcher, View view) {
        super();

        // Setting Data
        this.switcher = switcher;
        this.tag_selected = selected;
        this.tag_text = tag;

        // Finding Views
        text_view = view.findViewById(R.id.tag);

        // Setting Views
        text_view.setText(tag_text);
        text_view.setOnClickListener(this);
    }

    // Interfaces
    public interface TagSwitcher {
        void moveTag(String tag, boolean selected);
    }

    // Management Methods
    public void updateView(String tag, boolean selected) {
        this.tag_selected = selected;
        this.tag_text = tag;

        text_view.setText(tag_text);
    }

    // Click Listener
    @Override
    public void onClick(View view) {
        switcher.moveTag(tag_text, !tag_selected);
    }
}

File: app/src/main/java/jgappsandgames/smartreminderslite/holder/TagHolder.kt
package jgappsandgames.smartreminderslite.holder

// Views
import android.view.View
import android.widget.TextView

// App
import jgappsandgames.smartreminderslite.R

/**
 * TagHolder
 * Created by joshua on 12/26/2017.
 *
 * Class to Handle Tag Views in a list
 */
class TagHolder(private var tagText: String?, private var tagSelected: Boolean, private val switcher: TagSwitcher, view: View): View.OnClickListener {
    // Views ---------------------------------------------------------------------------------------
    private val textView: TextView = view.findViewById(R.id.tag)

    // Initializer ---------------------------------------------------------------------------------
    init {
        textView.text = tagText
        textView.setOnClickListener(this)
    }

    // Management Methods --------------------------------------------------------------------------
    fun updateView(tag: String, selected: Boolean) {
        this.tagSelected = selected
        this.tagText = tag

        textView.text = tagText
    }

    // Click Listeners -----------------------------------------------------------------------------
    override fun onClick(view: View) {
        switcher.moveTag(tagText, !tagSelected)
    }

    // Interfaces ----------------------------------------------------------------------------------
    interface TagSwitcher {
        fun moveTag(tag: String?, selected: Boolean)
    }
}